src/DIRAC/Core/Utilities/ElasticSearchDB.py

```diff
@@ -208,7 +200,7 @@ def getIndexPrefix(self):
         return self.__indexPrefix
 
     @ifConnected
-    def query(self, index, query):
+    def query(self, index: str, query: dict):
```

Comment:
This type hint isn't correct as it should always specify the type of the keys and value `dict[str, str]`. (There is also `TypedDict` for specifying key-specific types.)

Reply:
you mean it should be `dict[str, dict]`

Reply:
I don't know the opensearch API well enough to know what it should be, though it's important the type hints are either missing or correct else we'll have a mess trying to expand the use of `mypy`.

> you mean it should be `dict[str, dict]`

My original comment about needing types then applies to the second `dict` 😄 It might be this is a case for `dict[str, Any]` (though we should avoid using `Any` as much as possible). Alternatively we can not have a type hint at all and let mypy infer it from the opensearch bindings.

Reply:
OK got it. I removed the type hint for `body`